New behavior for ext power control

megamind4089 opened the pull request:
Added behaviors to control the external power output.

Tested with nRFMicro 1.3 and Lily58

Behaviors added:
&ext_power EP_ON
&ext_power EP_OFF
&ext_power EP_TOG

I am open to renaming things if there are better suggestions.

A big shoutout to @Tsquash , who helped me to test the behavior with nice!nano and nice!60.

innovaker commented:
Thanks @megamind4089!

A couple of Qs on this for @petejohanson:
- Should this be achieved with different behavior tags rather than using a flag?
- Can the shield enhancements be a separate commit? (same PR)

petejohanson commented:
> Thanks @megamind4089!
> 
> A couple of Qs on this for @petejohanson:
> - Should this be achieved with different behavior tags rather than using a flag?
> - Can the shield enhancements be a separate commit? (same PR)

Using the flag like is there is fine.

I'm not convinced we need the two parameters though. I would think for auto we would just pass `EP_AUTO` and that's that. Leaving as single parameter allows for using this with the hold-tap meta behavior.

innovaker commented:
Thanks!

I was wondering how we should differentiate between behavior tags and switches (flags) across all behaviors?  What convention should we encourage?  Together, they effectively mean "*do this, in this way*".  It could also be achieved with two tags belonging to the same behavior, effectively `&on` and `&off` - which also optimizes the keymap.

megamind4089 commented:
@innovaker I just used Lily58 to test this behavior and updated with this. I am happy to raise a new PR for shields also.

@petejohanson I have removed the second parameter to the behavior.
Also added toggle behavior as discussed

petejohanson reviewed (changes requested):
This is a great start on this. A few tweaks needed.

I think in general, using the "command parameter" like this is good, it avoids "driver instance proliferation", and leverages the argument infrastructure reasonably.

megamind4089 commented on app/src/behaviors/behavior_ext_power.c:
```diff
@@ -0,0 +1,59 @@
+        LOG_ERR("Unable to retrieve ext_power device: %d", command);
+        return -EIO;
+    }
+    const struct ext_power_api *ext_power_api = ext_power->driver_api;
```
What a shame I did. :cry:

petejohanson replied:
All good!

petejohanson commented on app/dts/bindings/behaviors/zmk,behavior-ext-power.yaml:
```diff
@@ -0,0 +1,10 @@
+
+compatible: "zmk,behavior-ext-power"
+
+include: two_param.yaml
```
Should be:

```suggestion
include: one_param.yaml
```

petejohanson commented on app/dts/behaviors/ext_power.dtsi:
```diff
@@ -0,0 +1,9 @@
+		ext_power: behavior_ext_power {
+			compatible = "zmk,behavior-ext-power";
+			label = "EXT_POWER_BEHAVIOR";
+			#binding-cells = <2>;
```
Doesn't need two:

```suggestion
			#binding-cells = <1>;
```

petejohanson reviewed (commented):
Couple minor tweaks.

petejohanson reviewed (approved):
@megamind4089 This is great! Can you please work on documenting the new behavior in a follow up?

Thanks again.
